.travis.yml

```diff
@@ -31,60 +24,60 @@ matrix:
         - DEPS=lowest
     - php: 5.6
       env:
-        - DEPS=latest
+        - DEPS=locked
         - TEST_COVERAGE=true
-        - DEPLOY_DOCS="$(if [[ $TRAVIS_BRANCH == 'master' && $TRAVIS_PULL_REQUEST == 'false' ]]; then echo -n 'true' ; else echo -n 'false' ; fi)"
-        - PATH="$HOME/.local/bin:$PATH"
+    - php: 5.6
+      env:
+        - DEPS=latest
     - php: 7
       env:
         - DEPS=lowest
     - php: 7
       env:
-        - DEPS=latest
+        - DEPS=locked
         - CS_CHECK=true
+    - php: 7
+      env:
+        - DEPS=latest
     - php: 7.1
       env:
         - DEPS=lowest
     - php: 7.1
       env:
+        - DEPS=locked
+    - php: 7.1
+      env:
         - DEPS=latest
-    - php: hhvm
+    - php: 7.2
       env:
         - DEPS=lowest
-    - php: hhvm
+    - php: 7.2
+      env:
+        - DEPS=locked
+    - php: 7.2
       env:
         - DEPS=latest
   allow_failures:
-    - php: hhvm
+    - php: 7.2
 
 before_install:
-  - if [[ $TRAVIS_PHP_VERSION != "hhvm" && $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini ; fi
+  - if [[ $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini || return 0 ; fi
   - travis_retry composer self-update
 
 install:
-  - travis_retry composer install $COMPOSER_ARGS
+  - travis_retry composer install $COMPOSER_ARGS --ignore-platform-reqs
```

Comment:
@weierophinney sure i understand how it works, but i dont think its usefull.

if we have a zf-module, which is needed in `zend-i18n` and which require PHP 7 in version 2 and someone create the lock file he have to add it in `$LEGACY_DEPS` because in version 1 the module works with PHP 5.6 too, or he have to add other special parts in the `travis.yml`.
We also dont check the require PHP modules, which maybe not installed, but which needed for some tests. Sure its possible to check in the tests if the module is installed, but if the module not exists, i think its not good to skip these tests.

i think its easier to generate a lock file on the lowest PHP version, which should work on all PHP versions.

correct me if im wrong, or if i missed somthing